### PR TITLE
test/k8sT: do not defer deletion of resources within It

### PR DESCRIPTION
**Pull request by ianvernon**
In K8sValidatedPolicyTestAcrossNamespaces, the deletion of pods and policies was wrapped with
`defer` calls, which meant that regardless of whether the test failed or succeeded, the
aforementioned resources were deleted within the `It` for the test. Destruction of
such resources should only be performed in the `AfterEach` block of the test, as we want
all resources to still be present for running log-gathering scripts like `cilium bugtool`.

Signed-off by: Ian Vernon <ian@cilium.io>

Related-to: #3623 (the logs were insufficient in that test run due to this issue).
Fixes: #3402

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -741,6 +742,12 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
 	})
 
 	AfterEach(func() {
+		for _, resource := range resources {
+			resourcePath := kubectl.ManifestGet(resource)
+			res := kubectl.Delete(resourcePath)
+			res.ExpectSuccess()
```
Comment (eloycoto) [inline test/k8sT/Policies.go]: I think that the ExpectSucess should not be called in AfterEach, if something fails here the following lines will not work. Happy if you decided to keep but I do not really like, assertions are for tests and JustAfterEach IMHO.

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -844,11 +848,11 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
 
 			By("Running tests WITH Policy / Proxy loaded")
 			testConnectivity(frontendPod.String(), backendSvcIP)
+			policyDeleteAndCheck(ciliumPods, policyPath, policyCmd)
```
Comment (eloycoto) [inline test/k8sT/Policies.go]: What happens if something fails in `testConnectivity`, the policy will never be deleted no? That function should be called on AfterEach or with a defer, if not the cleanup will never happen if the test failed.

File: test/k8sT/Policies.go [hunk 3]
```diff
@@ -857,6 +861,7 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
...
```
Comment (eloycoto) [inline test/k8sT/Policies.go]: What happens if something fails in testConnectivity, the policy will never be deleted no? That function should be called on AfterEach or with a defer, if not the cleanup will never happen if the test failed.

File: test/k8sT/Policies.go (hunk 1 above)
Comment (eloycoto) [reply test/k8sT/Policies.go]: If you want to keep this ExpectSucess should have an assert message.

**Review by eloycoto — DISMISSED** [review]

File: test/k8sT/Policies.go (hunk 1 above)
Comment (ianvernon) [reply test/k8sT/Policies.go]: I like this paradigm, makes sense to me.

File: test/k8sT/Policies.go (hunk 2 above)
Comment (ianvernon) [reply test/k8sT/Policies.go]: I'll make a call in `AfterEach` to delete the policy.

**Comment by ianvernon** [comment]
test-me-please

File: test/k8sT/Policies.go (hunk 3 above)
Comment (ianvernon) [reply test/k8sT/Policies.go]: Good call, fixed. Done in `AfterEach`.

File: test/k8sT/Policies.go (hunk 2 above)
Comment (ianvernon) [reply test/k8sT/Policies.go]: Good call, fixed. Done in `AfterEach`.

File: test/k8sT/Policies.go (hunk 1 above)
Comment (ianvernon) [reply test/k8sT/Policies.go]: Fixed.

**Review by eloycoto — APPROVED** [review]
LGTM

**Comment by ianvernon** [comment]
Will look into build failure: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2286/

Will test again to see if this is a flake.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Pull request merged**